corehq/apps/es/users.py

```diff
@@ -166,3 +167,13 @@ def role_id(role_id):
 
 def is_active(active=True):
     return filters.term("is_active", active)
+
+
+def user_data(key, value):
```

Comment:
Meant to ask - is this the best name for this filter?  Would it be helpful to call it `user_data_exact` or `filter_user_data` or something?

Reply:
Or, to throw in a wrench, what about `metadata`? HQ code doesn't access `user_data` directly, it's wrapped in the `metadata` property.

Reply:
Oh interesting - I didn't know about that change.  I'm actually not convinced, though, since it's not wrapped in ES code, and they're not quite the same thing.  This query only operates on the raw data sent to ES, so the run-time transformations that happen to `metadata` won't actually be queryable.  The results returned from ES will also contain only `user_data`, so by switching to `metadata`, there'd be a gap between what's searched and what's returned.  Is the intent to pretend that `user_data` doesn't exist?

Reply:
The metadata thing is a recent change, part of [user data profiles](https://github.com/dimagi/commcare-hq/pull/28208)

From a human behavior perspective, I want there to be as few references to the identifier `user_data` as possible, because it should no longer be accessed directly in python. In a world where couch migrations and ES reindexes weren't such a pain, I'd have renamed it `_user_data` to indicate it shouldn't be used, and probably also would have then renamed the value in the ES index.

But from the perspective of just this section of code, it is dealing with something called `user_data`, so there's logic to naming it similarly.

I agree it's messy either way and defer to your judgment.